weasyl/searchtag.py

```diff
@@ -73,6 +73,21 @@ def select_list(map_table, targetids):
     return dict(list(db.execute(q)))
 
 
+@region.cache_on_arguments()
+def get_or_create(name):
+    name = d.get_search_tag(name)
+    tag = d.engine.execute(
+        'INSERT INTO searchtag (title) VALUES (%(name)s) ON CONFLICT (title) DO NOTHING RETURNING tagid',
```

Comment:
Would an upsert style `ON CONFLICT (title) DO UPSERT` be a harmless noop here? It could save another query if so.